frontend/src/api/reflection/createReflection.ts

```diff
@@ -0,0 +1,294 @@
+import { API, graphqlOperation } from 'aws-amplify'
+import { GraphQLResult } from '@aws-amplify/api-graphql'
+
+import {
+  listMoments,
+  getReflexion,
+  listReflexions,
+} from '../../graphql/queries'
+
+import {
+  createReflexion,
+  createReflexionMoment,
+  updateReflexion,
+  deleteReflexionMoment,
+} from '../../graphql/mutations'
+import {
+  CreateReflexionInput,
+  UpdateReflexionInput,
+  CreateReflexionMomentInput,
+  DeleteReflexionMomentInput,
+  ReflexionState,
+  ListMomentsQuery,
+  Moment,
+  GetReflexionQuery,
+  GetMomentQueryVariables,
+  Reflexion,
+  ReflexionMoment,
+} from '../../API'
+
+export const updateReflexion2 = /* GraphQL */ `
+  mutation UpdateReflexion(
+    $input: UpdateReflexionInput!
+    $condition: ModelReflexionConditionInput
+  ) {
+    updateReflexion(input: $input, condition: $condition) {
+      id
+      createdAt
+      title
+      contentType
+      content
+      asset {
+        bucket
+        key
+        region
+      }
+      topic
+      subTopic
+      niveau
+      indicators
+      state
+      deleted
+      sharedUsers
+      comments {
+        createdAt
+        content
+      }
+      orientationQuestions {
+        question
+      }
+
+      moments {
+        items {
+          id
+          moment {
+            id
+          }
+        }
+        nextToken
+      }
+      updatedAt
+      owner
+    }
+  }
+`
+
+interface State extends CreateReflexionInput {
+  momentIDs?: string[]
+}
+
+export const createReflextionAPI = async (reflection: State) => {
```

Comment:
take a look here to make updates easier from the beginning. https://github.com/oseven-development/petrakip-app/blob/main/frontend/src/api/moment/saveMoment.ts